src/reducers/polling.js

```diff
@@ -0,0 +1,380 @@
+  type: UPDATE_POLL,
+  payload: { pollId, pollDataUpdates }
+});
+
+export const setOptionVotingFor = (pollId, optionId) => ({
+  type: POLLS_SET_OPTION_VOTING_FOR,
+  payload: { pollId, optionId }
+});
+
+// Writes ---
+
+export const voteForPoll = (pollId, optionId) => async dispatch => {
+  dispatch({ type: POLL_VOTE_REQUEST });
+
+  const pollVote = window.maker.service('govPolling').vote(pollId, optionId);
+  const success = await handleTx({
+    txObject: pollVote,
+    prefix: 'VOTE',
+    dispatch
+  });
+
+  if (success) {
+    dispatch(setOptionVotingFor(pollId, optionId));
+    dispatch(updateVoteBreakdown(pollId));
+  }
+};
+
+export const withdrawVoteForPoll = pollId => async dispatch => {
+  dispatch({ type: POLL_VOTE_REQUEST });
+
+  const pollVote = window.maker.service('govPolling').vote(pollId, 0);
+  const success = await handleTx({
+    txObject: pollVote,
+    prefix: 'VOTE',
+    dispatch
+  });
+
+  if (success) {
+    dispatch(setOptionVotingFor(pollId, 0));
+    dispatch(updateVoteBreakdown(pollId));
+  }
+};
+
+// Reads ---
+
+const getAllWhiteListedPolls = async () => {
+  const pollsList = await window.maker
+    .service('govPolling')
+    .getAllWhitelistedPolls();
+
+  const polls = uniqBy(pollsList, p => p.url);
+  console.log('unique polls', polls);
+  return polls;
+};
+
+export const getOptionVotingFor = (address, pollId) => async dispatch => {
+  const optionId = await window.maker
+    .service('govPolling')
+    .getOptionVotingFor(address, pollId);
+  dispatch(setOptionVotingFor(pollId, optionId));
+};
+
+const fetchPollFromUrl = async url => {
+  console.log('url to fetch', url);
+  const res = await fetch(url);
+  await check(res);
+  const json = await res.json();
+  if (!json['voteId']) {
+    return null;
+  } else {
+    console.log('valid poll', json);
+    return json;
+  }
+};
+
+const formatOptions = options => {
+  return Object.values(options);
+};
+
+const formatYamlToJson = data => {
+  const json = matter(data.about);
+  const { content } = json;
+  const { title, summary, options, discussion_link } = json.data;
+  return {
+    voteId: data.voteId,
+    title,
+    summary,
+    options: formatOptions(options),
+    discussion_link,
+    content,
+    rawData: data.about
+  };
+};
+
+const isPollActive = (startDate, endDate) => {
+  const now = new Date();
+  return startDate <= now && endDate > now ? true : false;
+};
+
+export const updateVoteBreakdown = pollId => (dispatch, getState) => {
+  const poll = getState().polling.polls.find(poll => poll.pollId === pollId);
+  if (!poll) return;
+  const { options, endDate } = poll;
+  async function checkForVoteBreakdownUpdates(triesRemaining) {
+    if (triesRemaining === 0) return;
+    const voteBreakdown = await getVoteBreakdown(pollId, options, endDate);
+    const totalVotes = await getTotalVotes(pollId);
+    const participation = await getParticipation(pollId);
+    const numUniqueVoters = await getNumUniqueVoters(pollId);
```

Comment:
@jparklev I liked the implementation here, so I tacked on some additional data for the Voter Status side bar section as well.